bin/setup_elastic.py

```diff
@@ -1,23 +1,24 @@
 from pyshelf.search.metadata import Metadata
-from elasticsearch import Elasticsearch
-from urlparse import urlparse
+from pyshelf.search import utils
 import yaml
 
 
 class ElasticInitializer(object):
-    def __init__(self, connection_string):
-        parsed = urlparse(connection_string)
-        url = parsed.scheme + "://" + parsed.netloc
-        self.index = parsed.path[1:]
-        self.es = Elasticsearch(url)
+    def __init__(self, config_path):
+        self.config_path = config_path
 
     def initialize(self):
-        Metadata.init(using=self.es, index=self.index)
-        self.es.indices.refresh(index=self.index)
+        config = self.read_config()
+        connection, index = utils.configure_es_connection(config["connectionString"],
+                config.get("accessKey"), config.get("secretKey"), config.get("region"))
+        Metadata.init(using=connection, index=index)
+        self.es.indices.refresh(index=index)
 
+    def read_config(self):
+        with open(self.config_path) as cf:
+            config = yaml.load(cf.read())
 
-with open("config.yaml") as cf:
-    config = yaml.load(cf.read())
+            return config.get("elasticsearch")
 
-elastic = ElasticInitializer(config.get("elasticSearchConnectionString"))
+elastic = ElasticInitializer("config.yaml")
```

Comment:
Could you find the absolute path based on the current file?  I think config.yaml only works if you execute it from the root directory.

```
directory_of_this_file = os.path.dirname(os.path.realpath(__file__))
config_path = os.path.realpath(os.path.join(direcory_of_this_file, "../config.yaml"))
```

Another thing to consider is allowing us to pass in a path and defaulting it to config.yaml if it isn't there.